lib/src/open_api/index.dart

```diff
@@ -59,6 +59,7 @@ bool _checkReferenceTypes(name, ref, self) {
       return true;
     } else if (self is SchemaObject) {
       // Reference types can be different if the reference is a SchemaObject
+      // should this return true?
```

Comment:
@walsha2 are you able to weigh in here at all?

I think the `_checkReferenceTypes` needs either a rework or a fix. 

The comment on this line seems to suggest it should return true.

Reply:
I have attempted to implement a fix, which works for my test cases.